test/LdapTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZendTest\Ldap;
+
+use phpmock\phpunit\PHPMock;
+use Zend\Ldap\Dn;
+use Zend\Ldap\Exception\LdapException;
+use Zend\Ldap\Ldap;
+
+class LdapTest extends \PHPUnit_Framework_TestCase
+{
+    use PHPMock;
+
+    public function setup()
+    {
+        if (!getenv('TESTS_ZEND_LDAP_ONLINE_ENABLED')) {
+            $this->markTestSkipped("Zend_Ldap online tests are not enabled");
+        }
+    }
+
+    /**
+     * @param $dn
+     * @param $attribs
+     * @param $sure
+     *
+     * @dataProvider removingAttributesProvider
+     */
+    public function testRemovingAttributes($dn, $attribs, $sure, $expectedDn, $expectedArray)
```

Comment:
* Please rename `$attribs` as `$attributes` for argument name match
* Please rename `$sure` as `$emptyAll` for argument name match
* Please rename `$expectedArray` as `$expectedAttributesToRemove` or something similar.